apps/sms/views/new_message/js/new_message.js

```diff
@@ -0,0 +1,3031 @@
+/* -*- Mode: js; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- /
+/* vim: set shiftwidth=2 tabstop=2 autoindent cindent expandtab: */
+
+/*global Compose, Recipients, Utils, Template, Settings,
+         SMIL, ErrorDialog, MessageManager, LinkHelper,
+         ActivityPicker, InboxView, OptionMenu, Threads, Contacts,
+         Attachment, WaitingScreen, MozActivity, LinkActionHandler,
+         TimeHeaders, ContactRenderer, Draft, Drafts,
+         MultiSimActionButton, Navigation, Promise, LazyLoader,
+         App,
+         SharedComponents,
+         ActivityClient,
+         Errors,
+         EventDispatcher,
+         SelectionHandler,
+         TaskRunner
+*/
+/*exported ConversationView */
+
+(function(exports) {
+'use strict';
+
+var attachmentMap = new WeakMap();
+
+function conv_mmsAttachmentClick(target) {
+  var attachment = attachmentMap.get(target);
+  if (!attachment) {
+    return false;
+  }
+
+  attachment.view({
+    allowSave: true
+  });
+
+  return true;
+}
+
+// reduce the Composer.getContent() into slide format used by SMIL.generate some
+// day in the future, we should make the SMIL and Compose use the same format
+function conv_generateSmilSlides(slides, content) {
+  var length = slides.length;
+  if (typeof content === 'string') {
+    if (!length || slides[length - 1].text) {
+      slides.push({
+        text: content
+      });
+    } else {
+      slides[length - 1].text = content;
+    }
+  } else {
+    slides.push({
+      blob: content.blob,
+      name: content.name
+    });
+  }
+  return slides;
+}
+
+var ConversationView = {
+  CHUNK_SIZE: 10,
+  // duration of the notification that message type was converted
+  CONVERTED_MESSAGE_DURATION: 3000,
```

Comment:
I'm not so sure the notification should move to Compose. I'd rather see a generalized notification system. This could be a very simple lib that would display a message in a notification bubble with some id (the id could be "calculated" from the notification name -- and with the notification name we'd have the duration using an array).

Just from the top of my head here; not sure this is the best solution. But moving to Compose is not a solution from my pov.